DOC-1849: Improve documentation of issue object and a11ychecker_filter_issue option (#2555)
tinymce/tinymce-docs: modules/ROOT/pages/a11ychecker.adoc

  }
]
----

[[issue]]
=== The `+issue+` object

> In a11ychecker, when the content within the editor is `audited`, each element is checked to ensure that no xref:a11ychecker.adoc#accessibility-rules[Accessibility rules] are violated. Any element which doesn't adhere to a rule will generate an `issue` within the audit; the details of which are to be displayed in the a11ychecker dialog.

Reviewer: ```suggestion
In a11ychecker, when the content within the editor is `audited`, each element is checked to ensure that no xref:a11ychecker.adoc#accessibility-rules[Accessibility rules] are violated. Any element which doesn’t adhere to a rule will generate an `issue` within the audit; the details of which are to be displayed in the a11ychecker dialog.
```

> The `+issue+` object is provided to any function implementation of this option. This object provides relevant data pertaining to the triggered issue:

Reviewer: ```suggestion
The `+issue+` object is provided to any option that implements said option as a function. This object provides relevant data pertaining to the triggered issue:
```

_Function_ here reads like it should be in the adjectival form, _functional_; describing the form taken by the noun phrase, _implementation of this option_.

I suspect that’s not true, but it’s hard to parse this sentence without thinking the noun is masquerading as a comparative.

The suggestion above is likely wrong but it’s me grasping at re-phrasing this so the original noun phrase and the noun-that-looks-reads-an-adjective can both still be nouns.

The reason I suspect it’s not right is that the re-written noun phrase is now, strictly speaking, a verb phrase with a referent — _implements said option_.

As I said, grasping.

Author: ```suggestion
The `+issue+` object provides relevant data pertaining to any issue generated by an element which violates an xref:a11ychecker.adoc#accessibility-rules[Accessibility rule]):
```
Ah, I missed this. I started writing this under the `a11ychecker_filter_issue` option, then found it already existed under the `getAudit()` API, so I moved the whole section to the bottom. It's an object which is generated in the audit (when the dialog is opened, or via the API), which is also provided to `a11ychecker_filter_issue` for filtering before the final audit is produced. Let me know if this sounds better.

Reviewer: This reads cleanly and succinctly.

* `+id+` : `+String+` A11ychecker issue identifier used by {productname}, such as D1, T4A. For a description and other details about the issue, see xref:accessibility-rules[Accessibility Rules].

> * `+description+` : `+String+` description of the issue, as will be displayed in the dialog.
> 
> * `+severity+` : `+String+` severity level of the issue, either `+info+`, `+warning+`, or `+error+`.

Reviewer: ```suggestion
* `+description+` : `+String+` description of the issue; as will be displayed in the dialog.

* `+severity+` : `+String+` severity level of the issue; either `+info+`, `+warning+`, or `+error+`.
```

A semi-colon to delimit the two descriptive clauses is a stronger demarcator and encourages the reader to take a mental breath between each clause.

Author: Good idea, thanks
